src/machine/machine_atsamd51.go

```diff
@@ -1331,12 +1331,13 @@ func (spi SPI) Configure(config SPIConfig) error {
 
 	// Determine the input pinout (for SDI).
 	var dataInPinout uint32
-	SDIPinMode, SDIPad, ok := findPinPadMapping(spi.SERCOM, config.SDI)
+	var SDIPinMode PinMode
```

Comment:
I think it's a good change.

If we don't use SDI, is it OK for SDIPinMode to be 0?
Do you know anything about it?

Reply:
For SDIPinMode, it's only used when config.SDI != NoPin so it's ok.

For dataInPinout, I concede I'm not 100% sure. What i can say is that my adafruit pygamer tft screen works seemlessly with this change.

Reply:
The proposed amendment to #1408 ([conejoninja@6c39859](https://github.com/conejoninja/tinygo/commit/6c39859ec32b0d28ce3aadd0f0966079fc5cc380)) is almost the same code.
I think it's fine.

Reply:
Yes I think this is fine. With `dataInPinout` left zero, it will still read data from some arbitrary pin but as long as it is not read that should be fine (see `CTRLA.DIPO` field of the SPI peripheral, [here](http://ww1.microchip.com/downloads/en/DeviceDoc/60001507E.pdf#page=986)).